azurelinuxagent/ga/exthandlers.py

```diff
@@ -2305,9 +2305,11 @@ def _parse_boolean_value(value, default_val):
         """
         Expects boolean value but
         for backward compatibility, 'true' (case-insensitive) is accepted, and other values default to False
+        Note: In py2, string can be unicode or str type. In py3, unicode removed
+        ustr is a unicode object for Py2 and a str object for Py3.
         """
         if not isinstance(value, bool):
-            return True if isinstance(value, str) and value.lower() == "true" else default_val
+            return True if isinstance(value, (ustr, str)) and value.lower() == "true" else default_val
```

Comment:
```suggestion
            return True if isinstance(value, (ustr, str)) and value.lower() == "true" else default_val
```
```suggestion
            return True if value.lower() == ustr("true") else default_val
```

Reply:
Initial intention was protecting against non-string/non-boolean and use default otherwise value.lower() will raise an error and will not process the extension.

Reply:
is your intention we are ok break the extension in that case? non-string/boolean totally unacceptable?